src/extension/app/components/plugin/plugin.js

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-disable max-len */
+
+import { html } from 'lit';
+import { EXTERNAL_EVENTS } from '../../constants.js';
+
+/**
+ * @typedef {import('@AppStore').AppStore} AppStore
+ */
+
+/**
+ * @typedef {import('@Types').CorePlugin} CorePlugin
+ */
+
+/**
+ * @typedef {import('@Types').CustomPlugin} CustomPlugin
+ */
+
+/**
+ * @typedef {import('../action-bar/picker/picker.js').Picker} Picker
+ */
+
+/**
+ * @typedef ContainerPlugin
+ * @property {Record<string, CorePlugin>} [children] The child plugins of the container
+ */
+
+/**
+ * The lit template result type
+ * @typedef {import('lit').TemplateResult} TemplateResult
+ */
+
+/**
+ * Creates a sidekick plugin.
+ * @param {CorePlugin|CustomPlugin} plugin The plugin configuration
+ */
+export class SidekickPlugin {
+  /**
+   * @type {Object<string, SidekickPlugin>}
+   */
+  children = {};
+
+  constructor(plugin) {
+    this.disabled = false;
+    this.config = plugin;
+    this.id = plugin.id;
+  }
+
+  /**
+   * Checks the plugin's condition.
+   * @returns {boolean} True if plugin's condition is met, else false
+   */
+  checkCondition() {
+    return !!this.config.condition?.(this.config.appStore);
```

Comment:
`condition?.(this.config.appStore)`

Wild, didn't know that was possible!

Reply:
Tried out a few different ways 😄